src/webportal/src/plugins/job-submission/components/App.jsx

```diff
@@ -54,38 +58,36 @@ export class App extends React.Component {
   }
 
   render() {
-    const {items} = this.state;
+    const {jobTaskRoles, parameters, jobInformation} = this.state;
     const formLayout = getFormClassNames().formLayout;
     const topForm = getFormClassNames().topForm;
 
     return (
       <Fabric>
         <Stack className={formLayout}>
           <Stack horizontal gap='l2'>
-            <Stack styles={{root: {width: '70%'}}} gap='l2'>
-              <Stack className={topForm}>
-                <JobInformation />
-              </Stack>
-              <Stack className={topForm} gap='l1'>
-                <Stack>
-                  <TabForm defaultItems={items}
-                           headerTextPrefix='Task Role'
-                           createContentFunc={() => new JobTaskRole({})}
-                           onRenderTabContent={this._onRenderTabContent.bind(this)}
-                           onItemsChange={(items)=>this.setState({items: items})}/>
-                </Stack>
-                <Stack horizontal gap='s1' horizontalAlign='center'>
-                  <PrimaryButton>Submit</PrimaryButton>
-                  <DefaultButton>Edit YAML</DefaultButton>
-                  <DefaultButton>Export</DefaultButton>
-                </Stack>
+            <StackItem grow styles={{root: {overflow: 'auto'}}}>
+              <Stack gap='l2'>
+                <StackItem className={topForm}>
+                  <JobInformation defaultValue={jobInformation}
+                                  onChange={(jobInformation) => this.setState({jobInformation: jobInformation})}/>
+                </StackItem>
+                <StackItem className={topForm}>
+                  <Stack gap='l1'>
+                    <TaskRoles defaultValue={[new JobTaskRole({})]}
+                              onChange={(jobTaskRoles) => this.setState({jobTaskRoles: jobTaskRoles})}/>
+                    <SubmissionSection job={new Job(jobInformation, jobTaskRoles, parameters)}/>
+                  </Stack>
+                </StackItem>
               </Stack>
-            </Stack>
-            <Stack styles={{root: {width: '30%'}}}>
+            </StackItem>
+            <StackItem disableShrink styles={{root: [t.w30]}}>
               <Stack className={topForm}>
-                <Parameter/>
+                <Parameters defaultValue={parameters}
+                            environment={[]}
+                            onChange={(parameters) => this.setState({parameters: parameters})}/>
```

Comment:
I think the `Parmeters` component is actually a controlled component since the `defaultValue` prop is mutable.